src/main/webapp/app/entities/competency/learning-path.model.ts

```diff
@@ -21,6 +21,29 @@ export class LearningPathInformationDTO {
     public progress?: number;
 }
 
+export enum LearningObjectType {
+    LECTURE = 'LECTURE',
+    EXERCISE = 'EXERCISE',
+}
+
+export class LearningPathNavigationObjectDto {
```

Comment:
I know we often use classes, but is there a reason we use classes over interfaces? 🤔 I think those could make more sense here.

Reply:
Makes completely sense here to switch to interfaces as we don't offer any methods and only need type checking